transport-interop/impl/js/v0.45/.aegir.js

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable no-console */
+import http from 'http'
+import { pEvent } from 'p-event'
+import { createClient } from 'redis'
+
+const redisAddr = process.env.redis_addr || 'redis:6379'
+const transport = process.env.transport
+const isDialer = process.env.is_dialer === 'true'
+
+/** @type {import('aegir/types').PartialOptions} */
+export default {
+  test: {
+    browser: {
+      config: {
+        // Ignore self signed certificates
+        browserContextOptions: { ignoreHTTPSErrors: true }
+      }
+    },
+    async before () {
+      // import after build is complete
+      const { createRelay } = await import('./dist/test/fixtures/relay.js')
+
+      let relayNode
+      let relayAddr
+      if (transport === 'webrtc' && !isDialer) {
+        relayNode = await createRelay()
+
+        const sortByNonLocalIp = (a, b) => {
+          if (a.toString().includes('127.0.0.1')) {
+            return 1
+          }
+          return -1
+        }
+
+        relayAddr = relayNode.getMultiaddrs().sort(sortByNonLocalIp)[0].toString()
+      }
+
+      const redisClient = createClient({
+        url: `redis://${redisAddr}`
+      })
+      redisClient.on('error', (err) => {
+        console.error('Redis client error:', err)
+      })
+      await redisClient.connect()
+
+      const requestListener = async function (req, res) {
+        const requestJSON = await new Promise(resolve => {
+          let body = ''
+          req.on('data', function (data) {
+            body += data
+          })
+
+          req.on('end', function () {
+            resolve(JSON.parse(body))
+          })
+        })
+
+        try {
+          const redisRes = await redisClient.sendCommand(requestJSON)
+
+          if (redisRes == null) {
+            console.error('Redis failure - sent', requestJSON, 'received', redisRes)
+
+            res.writeHead(500, {
+              'Access-Control-Allow-Origin': '*'
+            })
+            res.end(JSON.stringify({
+              message: 'Redis sent back null'
+            }))
+
+            return
+          }
+
+          res.writeHead(200, {
+            'Access-Control-Allow-Origin': '*'
+          })
+          res.end(JSON.stringify(redisRes))
+        } catch (err) {
+          console.error('Error in redis command:', err)
+          res.writeHead(500, {
+            'Access-Control-Allow-Origin': '*'
+          })
+          res.end(err.toString())
+        }
+      }
+
+      const proxyServer = http.createServer(requestListener)
+      proxyServer.listen(0)
```

Comment:
This is what fixed it. Previously this [was](https://github.com/libp2p/js-libp2p/blob/libp2p-v0.46.21/interop/.aegir.js#L71):

```
  await new Promise(resolve => { proxyServer.listen(0, 'localhost', () => { resolve() }) })
```

The reason this broke it is kind of funny. For whatever reason this now listens on only the IPv6 loopback address `::1`, but when we do the fetch:

```
 fetch(`http://localhost:...
 ```
 
 it resolves to the ipv4 loopback address at `127.0.0.1`.
 
 ---
 
 Apparently in this environment just doing `.listen(0)` listens on both `::1` and `127.0.0.1` (unless IPv6 is disabled and you only listen on 127.0.0.1).

Reply:
Yeah, it's generally better to listen without specifying a host, then it'll bind to all available addresses/interfaces, then the host can resolve `"localhost"` to whatever it wants and we can have a quiet life 😄